File: grocery-manager-backend/src/main/java/com/grocerymanager/api/model/StoreLocation.java
package com.grocerymanager.api.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "store_locations")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class StoreLocation {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank
    @Size(max = 255)
    private String name;

    @NotBlank
    @Size(max = 500)
    private String address;

    @NotNull
    private Double latitude;

    @NotNull
    private Double longitude;

    @NotBlank
    @Size(max = 100)
    private String geofenceId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    @ToString.Exclude
    private User user;

    @CreationTimestamp
    private LocalDateTime createdAt;

    @UpdateTimestamp
    private LocalDateTime updatedAt;

    private String syncId;
    private LocalDateTime lastSynced;

    // For managing synchronization conflicts
    @Version
    private Long version;
}

File: frontend/app/src/main/java/com/example/frontend/data/model/StoreLocation.kt
package com.example.frontend.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.time.LocalDateTime

@Entity(tableName = "store_locations")
data class StoreLocation(
    @PrimaryKey(autoGenerate = true)
    val id: Int = 0,
    val name: String,
    val address: String,
    val latitude: Double,
    val longitude: Double,
    val geofenceId: String,
    val syncId: String? = null,
    val serverId: Long? = null,
    val updatedAt: LocalDateTime? = LocalDateTime.now(),  // Update date
    val syncStatus: SyncStatus = SyncStatus.LOCAL_ONLY
)